external-crates/move/move-compiler/src/unit_test/plan_builder.rs

```diff
@@ -330,9 +330,18 @@ fn parse_failure_attribute(
                             value_name_loc,
                             &attr_value,
                         )?;
+                    if let Some((module_id, const_name)) = &const_location_opt {
+                        // we found a constant used in the `expected_failure` annotation - if a warning about
+                        // constant being unused was generated, remove it
+                        context.env.remove_unused_const_diag(
+                            &module_id.name().as_str().into(),
+                            const_name,
+                        );
+                    }
```

Comment:
This seems like a weird way to adjust test behaviors. If anything, I think you'd check this at error generation time in the actual computation code. I can imagine that being preferable for other reasons, such as introducing an `#[ignore-unused]` annotation or similar for constants we'd like to allocate for future-proofing, etc.

Reply:
I am adopting @tnowacki's refactoring suggestion and this will change as a result to avoid the weirdness.